kapitan/inputs/base.py

```diff
@@ -118,12 +118,15 @@ def write_yaml(self, obj):
         else:
             obj = self.revealer.compile_obj(obj, target_name=target_name)
 
-        if isinstance(obj, Mapping):
-            yaml.dump(obj, stream=self.fp, indent=indent, Dumper=PrettyDumper, default_flow_style=False)
-        else:
-            yaml.dump_all(obj, stream=self.fp, indent=indent, Dumper=PrettyDumper, default_flow_style=False)
+        if obj:
+            if isinstance(obj, Mapping):
+                yaml.dump(obj, stream=self.fp, indent=indent, Dumper=PrettyDumper, default_flow_style=False)
+            else:
+                yaml.dump_all(obj, stream=self.fp, indent=indent, Dumper=PrettyDumper, default_flow_style=False)
 
-        logger.debug("Wrote %s", self.fp.name)
+            logger.debug("Wrote %s", self.fp.name)
+        else:
+            logger.debug("%s is None, didn't write", self.fp.name)
```

Comment:
Maybe "%s is Empty, skipped writing output" instead?

Reply:
Done